massa-execution-worker/src/execution.rs

```diff
@@ -1354,4 +1406,95 @@ impl ExecutionState {
     pub fn get_address_future_deferred_credits(&self, address: &Address) -> BTreeMap<Slot, Amount> {
         context_guard!(self).get_address_future_deferred_credits(address, self.config.thread_count)
     }
+
+    /// Get the execution statuses of both speculative and final executions
+    ///
+    /// # Return
+    ///
+    /// * A tuple of hashmaps with:
+    /// * first the statuses for speculative executions
+    /// * second the statuses for final executions
+    pub fn get_op_exec_status(&self) -> (HashMap<OperationId, bool>, HashMap<OperationId, bool>) {
+        (
+            self.active_history.read().get_op_exec_status(),
+            self.final_state.read().executed_ops.op_exec_status.clone(),
+        )
+    }
+
+    /// Initialize the hashmap of addresses from the vesting file
+    pub fn init_vesting_registry(
+        config: &ExecutionConfig,
+    ) -> Result<PreHashMap<Address, Vec<VestingRange>>, ExecutionError> {
+        let mut hashmap: PreHashMap<Address, Vec<VestingRange>> = serde_json::from_str(
+            &std::fs::read_to_string(&config.initial_vesting_path).map_err(|err| {
+                ExecutionError::InitVestingError(format!(
+                    "error loading initial vesting file  {}",
+                    err
+                ))
+            })?,
+        )
+        .map_err(|err| {
+            ExecutionError::InitVestingError(format!(
+                "error on deserialize initial vesting file  {}",
+                err
+            ))
+        })?;
+
+        let get_slot_at_timestamp = |config: &ExecutionConfig, timestamp: MassaTime| {
+            match massa_models::timeslots::get_latest_block_slot_at_timestamp(
+                config.thread_count,
+                config.t0,
+                config.genesis_timestamp,
+                timestamp,
+            ) {
+                Ok(opts) => Ok(opts.unwrap()),
+                Err(_) => Err(ExecutionError::InitVestingError(format!(
+                    "can no get the slot at timestamp : {}",
+                    timestamp
+                ))),
+            }
+        };
+
+        for v in hashmap.values_mut() {
+            if v.len().eq(&1) {
+                return Err(ExecutionError::InitVestingError(
+                    "vesting file should has more one element".to_string(),
```

Comment:
nit: "vesting file should _have_ more _than_ one element"


...just minor grammar

Reply:
...or "vesting file should have at least one element"